packages/components/src/components/tab/group.tsx

```diff
@@ -78,12 +78,12 @@ export const TabGroup = () => {
           onClick={() => {
             const newTab = createNewDefaultTab(totalTabsCreated)
             const newTabs: Tab[] = [...tabs, newTab]
-            setTabs(newTabs)
+            setTabs(() => newTabs)
             setTotalTabsCreated((totalTabsCreated) => totalTabsCreated + 1)
             updateCurrentTabId(newTab.id)
```

Comment:
this is some really bad state management code I wrote here. But it's because I had a lot of trouble with getting the tabs to work (changing their positions and keeping their state intact). If you think you can rewrite it to make it better, then feel free to do that.

Reply:
I also found some issues with the tabs state, when creating a new tab there sometimes occurs a collision of id's.
I think this can we solved by storing the `totalTabsCreated` in localStorage or we could get rid of `totalTabsCreated` and just create a random id when a new tab is created.

What do you think?

Reply:
I think the random id is a good solution for now